wgpu-types/src/lib.rs

```diff
@@ -100,6 +100,24 @@ impl From<Backend> for BackendBit {
     }
 }
 
+/// This type is not to be constructed by any users of wgpu. If you construct this type, any semver
+/// guarantees made by wgpu are invalidated and a non-breaking change may break your code.
+///
+/// If you are here trying to construct it, the solution is to use partial construction with the
+/// default:
+///
+/// ```ignore
+/// let limits = Limits {
+///     max_bind_groups: 2,
+///     ..Limits::default()
+/// }
+/// ```
+#[doc(hidden)]
+#[derive(Debug, Copy, Clone, Default, Eq, PartialEq, Hash)]
+#[cfg_attr(feature = "trace", derive(Serialize))]
+#[cfg_attr(feature = "replay", derive(Deserialize))]
+pub struct NonExhaustive;
```

Comment:
Can we make it so the type is opaque and the only way to create it is using an unsafe constructor?
This way, users have to at least go through `unsafe()` if they decide that they want to use the type